### ISSUE TITLE
cf app can't send a request to open api. 

### ISSUE THREAD

**Issue by SimyungYang** [cloudfoundry/cf-release#940]
Now, I'm running and testing CF.
When one of my node app tried to send a request to open API, I've got this error message.

"502 Bad Gateway: Registered endpoint failed to handle the request."

I think outbound packet is stuck in somewhere, I'm not sure though.

I checked the  running security groups, space security groups.
It seems okay. running security groups has public_networks and it has the following configuration.

`
Name    public_networks   
Rules      
	[
		{
			"destination": "0.0.0.0-9.255.255.255",
			"protocol": "all"
		},
		{
			"destination": "11.0.0.0-169.253.255.255",
			"protocol": "all"
		},
		{
			"destination": "169.255.0.0-172.15.255.255",
			"protocol": "all"
		},
		{
			"destination": "172.32.0.0-192.167.255.255",
			"protocol": "all"
		},
		{
			"destination": "192.169.0.0-255.255.255.255",
			"protocol": "all"
		}
	]
`

And I also check the firewall. After I ssh into runner vm, and run a curl command to that url.
It worked. So this means there is no firewall issue.

After I ran into this problem, I also tested with java app. and it has same problem.
Any ideas or suggestion?

======================

I ssh into warden container, and run a curl command. I can't get a correct response.
I guess it's warden security permission. But I couldn't what manifest configuration cause this issue.

**Comment by robdimsdale** [comment]
@SimyungYang it sounds like you're using warden, so let's see if the team responsible for that runtime has any thoughts.

/cc @fraenkel @jberkhahn @swetharepakula

**Comment by swetharepakula** [comment]
Hi @SimyungYang,

We wanted to get some more information to help you debug the issue. Could you see what your manifest says for warden.allow_networks? Also inside the runner vm could you send us the output of `iptables --list`. That will help us determine if this is cause by some warden configuration or not.

Thanks,
@swetharepakula, @jberkhahn 
RuntimeOG Team

**Comment by fraenkel** [comment]
@SimyungYang Are you using Diego or DEAs?  There is no SSH to DEA containers so I would think you are using Diego.

How did you deploy this Open API? Is it a CF app?
Who is returning the 502 error? Usually the headers will provide additional details.

**Comment by SimyungYang** [comment]
I am using DEA version. and SSH into the container by the following method.
https://myupbeat.wordpress.com/2015/05/09/ssh-into-the-warden-containers-running-within-a-dea-in-bosh-lite/

Open API meant Cloud Foundry API : http://apidocs.cloudfoundry.org/233/.

And dea_next.allow_networks key is confugured as 0.0.0.0/0.
Its value had nothing at first, and I changed it to the above when I ran into this issue.
Although I set the the above configuration, I still have the issue.

here is the result of `iptables --list`.

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     all  --  anywhere             anywhere             state RELATED,ESTABLISHED /* related-traffic */
REJECT     all  --  10.254.0.0/22        anywhere             /* block-traffic-to-host */ reject-with icmp-port-unreachable

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         
warden-forward  all  --  anywhere             anywhere            

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain warden-default (7 references)
target     prot opt source               destination         
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
RETURN     all  --  anywhere             anywhere            
REJECT     all  --  anywhere             anywhere             reject-with icmp-port-unreachable

Chain warden-forward (1 references)
target     prot opt source               destination         
ACCEPT     all  --  anywhere             anywhere            
warden-i-19f7c8q6prj  all  --  anywhere             anywhere            [goto] 
warden-i-19f7c8q6prf  all  --  anywhere             anywhere            [goto] 
warden-i-19f7c8q6pr5  all  --  anywhere             anywhere            [goto] 
warden-i-19f7c8q6pr9  all  --  anywhere             anywhere            [goto] 
warden-i-19f7c8q6pra  all  --  anywhere             anywhere            [goto] 
warden-i-19f7c8q6prb  all  --  anywhere             anywhere            [goto] 
warden-i-19f7c8q6pr6  all  --  anywhere             anywhere            [goto] 
DROP       all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6pr5 (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6pr5-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6pr5 "
RETURN     all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6pr6 (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6pr6-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6pr6 "
RETURN     all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6pr9 (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6pr9-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6pr9 "
RETURN     all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6pra (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6pra-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6pra "
RETURN     all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6prb (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6prb-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6prb "
RETURN     all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6prf (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6prf-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6prf "
RETURN     all  --  anywhere             anywhere            

Chain warden-i-19f7c8q6prj (1 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             10.0.2.0/24         
RETURN     all  --  anywhere             10.0.1.0/24         
RETURN     all  --  anywhere             10.0.0.0/24         
RETURN     udp  --  anywhere             anywhere             udp dpt:domain
RETURN     tcp  --  anywhere             anywhere             tcp dpt:domain
RETURN     all  --  anywhere             anywhere             destination IP range 52.0.0.0-52.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 192.169.0.0-255.255.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 172.32.0.0-192.167.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 169.255.0.0-172.15.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.1.0-10.0.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.0.2.0-10.0.2.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.1.0-10.10.1.255
RETURN     all  --  anywhere             anywhere             destination IP range 10.10.0.0-169.253.255.255
RETURN     all  --  anywhere             anywhere             destination IP range 0.0.0.0-9.255.255.255
warden-default  all  --  anywhere             anywhere            [goto] 

Chain warden-i-19f7c8q6prj-log (0 references)
target     prot opt source               destination         
LOG        tcp  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED LOG level warning prefix "warden-i-19f7c8q6prj "
RETURN     all  --  anywhere             anywhere            

```

**Comment by SimyungYang** [comment]
It's not a outbound problem. 

I deployed another app in PWS just to check out that the request is correctly received in the server.
And the request was well received in the internet server.

**the problem is that the response packet is not delivered to the cf apps.**

I guess It may be gorouter issue, or my IaaS's.

**Comment by fraenkel** [comment]
@SimyungYang X-Cf-Routererror:endpoint_failure means the router failed to access the endpoint. You are obviously making it out of the container.  Where do you think you are going?

**Comment by SimyungYang** [comment]
@fraenkel i guess it might not be a cf issue. i am not sure though. i running cf on openstack community version. when i check the packet on openstack with tcpdump, the response pacjet is not passing through between neutron and nova.
i will comment if i find more detail. thank you

**Comment by fraenkel** [comment]
@SimyungYang We are closing this for now. You can reopen it if you find out more details.

**Comment by SimyungYang** [comment]
I finally figured out this problem. It was due to MTU size of CF.
From cf docs, when i use GRE tunneling in openstack, MTU size is recommended as 1454. 
My environment has used vxlan, and CF works in 1450 or less MTU size.

I tested through the docker container on my openstack, and it also has the same problem.
So I googled about it and some guy said MTU may be the reason.

Thank you guys for help.
